source/material_model/steinberger.cc

```diff
@@ -489,6 +529,53 @@ namespace aspect
               seismic_out->vs[i] = seismic_Vs(in.temperature[i], in.pressure[i], in.composition[i], in.position[i]);
             }
         }
+
+      /* We separate the calculation of specific heat and thermal expansivity,
+        * because they may depend on cell-wise averaged values that are only
+        * available here.
+        */
+      double average_temperature(0.0);
+      double average_density(0.0);
+      for (unsigned int i = 0; i < in.position.size(); ++i)
+        {
+          average_temperature += in.temperature[i];
+          average_density += out.densities[i];
+        }
+      average_temperature /= in.position.size();
+      average_density /= in.position.size();
+
+      std_cxx1x::array<std::pair<double, unsigned int>,2> dH;
+      if (latent_heat && in.current_cell.state() == IteratorState::valid)
+        dH = enthalpy_derivative(in);
+
+      for (unsigned int i = 0; i < in.position.size(); ++i)
+        {
+          if (latent_heat)
+            {
+              // Use the adiabatic pressure instead of the real one, because of oscillations
+              const double pressure = this->get_adiabatic_conditions().pressure(in.position[i]);
+
+              if ((in.current_cell.state() == IteratorState::valid)
+                  && (dH[0].second > 0)
+                  && (dH[1].second > 0))
+                {
+                  out.thermal_expansion_coefficients[i] = (1 - average_density * dH[1].first) / average_temperature;
+                  out.specific_heat[i] = dH[0].first;
+                }
+              else
+                {
+                  if (material_lookup.size() == 1)
```

Comment:
Is the case above implemented for `material_lookup.size() > 1`? If not, it might be easier to just make an Assert out of that if statement and put it somewhere above.